python/pyspark/sql/functions.py

```diff
@@ -149,7 +152,17 @@ def lit(col: Any) -> Column:
     +------+---+
     |     5|  0|
     +------+---+
+
+    Create a literal from a list.
+
+    >>> spark.range(1).select(F.lit([1, 2, 3])).show()
+    +--------------+
+    |array(1, 2, 3)|
+    +--------------+
+    |     [1, 2, 3]|
+    +--------------+
     """
+    col = array(*[lit(item) for item in col]) if isinstance(col, list) else col
```

Comment:
Actually we should probably block the case of `[Column, Column]` because Scala side does not allow that case.

Reply:
so may looks like this:
```
    if isinstance(col, Column):
        return col
    elif isinstance(col, list):
        if any(isinstance(c, Column) for c in col):
            raise ValueError(...)
        return array(*[lit(item) for item in col])
    else:
        return _invoke_function("lit", col)
```

Reply:
qq: Maybe seems like the Scala side even doesn't allow for the single `Column` case ??

```scala
scala> df.select(lit(df.id)).show()
<console>:24: error: value id in class Dataset cannot be accessed in org.apache.spark.sql.Dataset[Long]
       df.select(lit(df.id)).show()
```

which is allowed in Python side:

```python
>>> df.select(F.lit(df.id)).show()
+---+
| id|
+---+
|  0|
|  1|
|  2|
|  3|
|  4|
|  5|
|  6|
|  7|
|  8|
|  9|
+---+
```

Reply:
df.id isn't a way to access a column in scala. try either $"id" or df.col("id")

Reply:
Thanks!